content/200-concepts/050-overview/300-prisma-in-your-stack/03-is-prisma-an-orm.mdx

```diff
@@ -222,50 +164,123 @@ The drawbacks of ORMs are not always apparent when you start using them. This se
 
-Models in Prisma mean something slightly different to ORMs. When using ORMs, models are represented as _classes_. With Prisma, models are defined in the Prisma schema as abstract entities which describe the relationships between tables:
+### Prisma schema
+
+At the heart of Prisma's implementation of the Data Mapper pattern is the _Prisma schema_ – a single source of truth for the following responsibilities:
+
+- Configuring how Prisma connects your database.
+- Generating Prisma Client – the type-safe ORM for use in your application code.
+- Creating and evolving the database schema with Prisma Migrate.
+- Defining the mapping between application objects and database columns.
+
+Models in Prisma mean something slightly different to Active Record ORMs. With Prisma, models are defined in the Prisma schema as abstract entities which describe tables, relations, and the mappings between columns to properties in the Prisma Client.
+
+As an example, here's a Prisma schema for a blog:
 
 ```prisma
-model User {
-  id        Int      @default(autoincrement()) @id
-  birthDate DateTime
-  email     String   @unique
-  firstName String?
-  lastName  String?
-  posts     Post[]
+datasource db {
+  provider = "postgres"
+  url      = "postgresql://janedoe:janedoe@localhost:5432/hello-prisma"
+}
+
+generator client {
+  provider = "prisma-client-js"
 }
 
 model Post {
-  content   String?
-  post_id   Int     @default(autoincrement()) @id
+  id        Int     @id @default(autoincrement())
   title     String
-  author_id User
+  content   String? @map("post_content")
+  published Boolean @default(false)
+  author    User?   @relation(fields: [authorId], references: [id])
+  authorId  Int?
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+  posts Post[]
 }
 ```
 
-Prima schema's declarative nature is concise compared to imperative model definitions in ORMs.
+Here's a break down of the example above:
+
+- The `datasource` block defines the connection to the database.
+- The `generator` block tells Prisma to generate the client for TypeScript and Node.js.
+- The `Post` and `User` models map to database tables.
+- The two models have a _1-n_ relation where each `User` can have many related `Post`s.
+- Each field in the models has a type, e.g. the `id` has the type `Int`.
+- Fields may contain field attributes to define:
+  - Primary keys with the `@id` attribute.
+  - Unique keys with the `@unique` attribute.
+  - Default values with the `@default` attribute.
+  - Mapping between table columns and Prisma Client fields with the `@map` attribute, e.g., the `content` field maps (which will be accessible in Prisma Client) maps to the `post_content` database column.
+
+The `User` / `Post` relation can be visualized with the following diagram:
+
+![1-n relation between User and Post](https://imgur.com/lidq1Xy.png)
```

Comment:
Database: Can `published` really be a String to be mapped to a Boolean?

Reply:
Nope. I've corrected that.